cmd/exchanger/main.go

```diff
@@ -9,53 +9,52 @@ import (
 
 	"github.com/ethanjmarchand/exchanger/internal/controller"
 	"github.com/ethanjmarchand/exchanger/internal/currency"
-	"github.com/go-chi/chi"
-	"github.com/go-chi/chi/middleware"
 	"github.com/joho/godotenv"
 )
 
-// TODO: Test this funtion.
-// Mixing concerns. This should only load an Env variable. Should be returning a string, or config struct.
-func loadEnvKey() (string, error) {
+// loadEnv simple grabs our env variables from our .env file, and returns a config.
+func loadEnv() (*currency.Config, error) {
+	config := currency.Config{}
 	err := godotenv.Load()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	CCKey := os.Getenv("CCKEY")
-	// Check to see if CCKEY is empty string.
-	if CCKey == "" {
-		return "", errors.New("CCKey cannot be an empty string.")
+	config.CCKey = os.Getenv("CCKEY")
+	config.URL = os.Getenv("APIURL")
+	if config.CCKey == "" {
+		return nil, errors.New("cckey cannot be an empty string")
 	}
-	return CCKey, nil
+	return &config, nil
 }
 
-func run(CCKey string) error {
-	currencyService, err := currency.NewService(CCKey)
+// run is our wrapper to check for errors and to hopefully better test.
+func run(config currency.Config) error {
+	currencyService, err := currency.NewService(config)
 	if err != nil {
 		return err
 	}
 	conv := controller.Converter{
 		CS: &currencyService,
 	}
-	// TODO: Get rid of chi.
-	r := chi.NewRouter()
-	r.Use(middleware.Logger)
-	r.Get("/", controller.Static)
-	r.Get("/exchange/{have}/{want}", conv.Render)
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("GET /", controller.Static)
+	mux.HandleFunc("GET /exchange/{have}/{want}", conv.Render)
 	fmt.Println("Server starting on port :3000...")
-	err = http.ListenAndServe(":3000", r)
+	err = http.ListenAndServe(":3000", mux)
```

Comment:
nit: It might be good for the port to come from the env variables so it can be changed easily without having to recompile code